src/uwtools/tests/config/formats/test_base.py

```diff
@@ -127,6 +128,21 @@ def test_dereference(caplog, config):
     assert config == {"foo": 88, "a": 77, "b": {"c": 66}, "d": "{{ X }}"}
 
 
+@pytest.mark.parametrize("fmt2", [FORMAT.ini, FORMAT.nml, FORMAT.sh])
+def test_invalid_config(caplog, fmt2, tmp_path):
+    """
+    Test that invalid config files will error when attempting to dump.
+    """
+    fmt1 = FORMAT.yaml
+    outfile = tmp_path / f"test_{fmt1}to{fmt2}_dump.{fmt2}"
+    cfgin = tools.format_to_config(fmt1)(fixture_path("hello_workflow.yaml"))
+    with raises(UWConfigError) as e:
+        tools.format_to_config(fmt2).dump_dict(path=outfile, cfg=cfgin.data)
+    msg = f"Cannot dump depth-8 config to type-'{fmt2}' config"
```

Comment:
I think you might consider here and in later similar tests improving the maintainability of the messages.

If you use the tool we have to check the depth of `cfgin` and have that dynamically included in the string, the tests will be more resilient to changes in our fixture files that are irrelevant to this test. If we changed it so much that it was compatible with config it's tested against, the test would fail and we'd maybe choose a new fixture file.

Reply:
Good point! I've updated it to pull `depth(cfgin.data)` in both sets of tests, thats for the best - thank you!